SharpSource/SharpSource/Diagnostics/DateTimeNowAnalyzer.cs

```diff
@@ -26,18 +24,15 @@ public override void Initialize(AnalysisContext context)
     {
         context.EnableConcurrentExecution();
         context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
-        context.RegisterSyntaxNodeAction(AnalyzeSymbol, SyntaxKind.SimpleMemberAccessExpression);
+        context.RegisterOperationAction(AnalyzePropertyReferences, OperationKind.PropertyReference);
     }
 
-    private static void AnalyzeSymbol(SyntaxNodeAnalysisContext context)
+    private static void AnalyzePropertyReferences(OperationAnalysisContext context)
     {
-        var expression = (MemberAccessExpressionSyntax)context.Node;
-
-        if (context.SemanticModel.GetSymbolInfo(expression.Expression).Symbol is INamedTypeSymbol symbol &&
-            symbol.SpecialType == SpecialType.System_DateTime &&
-            expression.Name.Identifier.ValueText == "Now")
+        var propertyReference = (IPropertyReferenceOperation)context.Operation;
+        if (propertyReference is { Type: { SpecialType: SpecialType.System_DateTime }, Property: { Name: "Now" } })
```

Comment:
This should work as you are using C# 10.0

```suggestion
        if (propertyReference is { Type.SpecialType: SpecialType.System_DateTime, Property.Name: "Now" })
```

Note that another approach is:

```csharp
// in Initialize
context.RegisterCompilationStartAction(context =>
{
    var dateTimeSymbol = context.Compilation.GetSpecialType(SpecialType.System_DateTime);
    var nowPropertySymbol = dateTimeSymbol.GetMembers("Now").OfType<IPropertySymbol>().FirstOrDefault();
    if (nowPropertySymbol is not null)
    {
        context.RegisterOperationAction(context => AnalyzePropertyReference(context, nowPropertySymbol), OperationKind.PropertyReference);
    }
});

// in AnalyzerPropertyReference, you can do a symbol comparison.
private static void AnalyzePropertyReference(OperationAnalysisContext context, IPropertySymbol nowProperty)
{
    var propertyReference = (IPropertyReferenceOperation)context.Operation;
    if (nowProperty.Equals(propertyReference.Property, SymbolEqualityComparer.Default))
    {
        context.ReportDiagnostic(Diagnostic.Create(Rule, propertyReference.Syntax.GetLocation()));
    }
}
```

I'm writing through GitHub directly, so I could have written something wrong.

Reply:
I like that approach 👍 Changed it